commands/local_run.go

```diff
@@ -82,13 +90,28 @@ services deployed within your OpenFaaS cluster.`,
 	}
 
 	cmd.Flags().BoolVar(&opts.print, "print", false, "Print the docker command instead of running it")
+	cmd.Flags().BoolVar(&opts.build, "build", true, "Build function prior to local-run")
 	cmd.Flags().IntVarP(&opts.port, "port", "p", 8080, "port to bind the function to")
 	cmd.Flags().StringVar(&opts.network, "network", "", "connect function to an existing network, use 'host' to access other process already running on localhost. When using this, '--port' is ignored, if you have port collisions, you may change the port using '-e port=NEW_PORT'")
 	cmd.Flags().StringToStringVarP(&opts.extraEnv, "env", "e", map[string]string{}, "additional environment variables (ENVVAR=VALUE), use this to experiment with different values for your function")
 
 	return cmd
 }
 
+func localBuild(cmd *cobra.Command, args []string) error {
+	err := preRunBuild(cmd, args)
```

Comment:
I think I'd probably inline these with `if err:= .... ; err !=nil {`

How does this work when there are flags that the build command uses like `--build-arg` - which are not present in the local run command's flags?

I.e. `faas-cli build --build-arg GO111MODULE=on`

Have a look at how `faas-cli up` works

Reply:
Made changes